src/pymodaq/utils/parameter/utils.py

```diff
@@ -52,7 +133,8 @@ def iter_children(param, childlist=[]):
     """
     for child in param.children():
         childlist.append(child.name())
-        if 'group' in child.type():
+        if child.hasChildren():
```

Comment:
ok

Reply:
_I don't like the fact to modify the widget depending on what is in your action_list. I would leave the ParameterWidget as it is with all actions. Then depending the action list in the ParameterManager, We would show (or not) the corresponding action (button) using a setVisible method. It means also the connection will always be there between the button and the corresponding method but is the button is not visible, the user cannot trigger the action so we're fine. Like this the object stay always the same, it is just we are allowing or not some of its features. And by default, I would allow all actions (in the init)_

You are right, I did not make full use of the add_action function which allows to define if it should be visible or not.
I added it in the setup_actions, the button is only showed if its associated action is given in the action list